### PR TITLE
feat: allow configure diode producer_app_name prefix

### PR DESCRIPTION
**Pull request by leoparente**
also fix issues:
- device-discovery: not integer port number 
- device-discovery:  not applying resolved api_key env variable

### CODE REVIEW

File: device-discovery/device_discovery/main.py [hunk 1]
```diff
@@ -62,6 +62,14 @@ def main():
         required=True,
     )
 
+    parser.add_argument(
+        "-a",
+        "--diode-prefix-name",
```
Comment (leoparente) [inline device-discovery/device_discovery/main.py]: @mfiedorowicz should it be:
diode-name-prefix
diode-app-name-prefix

or the way that it is? 🤔

Comment (mfiedorowicz) [reply device-discovery/device_discovery/main.py]: Leaning towards `diode-app-name-prefix`

Comment (leoparente) [reply device-discovery/device_discovery/main.py]: thanks, done

**Review by mfiedorowicz — APPROVED** [review]
🚀

**Pull request merged**